File: 3501-delete-nodes-from-linked-list-present-in-array/3501-delete-nodes-from-linked-list-present-in-array.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* modifiedList(int* nums, int numsSize, struct ListNode* head) {
    struct ListNode* dummy = (struct ListNode*)malloc(sizeof(struct ListNode));
    dummy->next = head;
    struct ListNode* ans = dummy;
    bool map[100001] = {0};
    while (numsSize--) map[*nums++] = 1;
    while (head) {
        if (map[head->val]) dummy->next = head->next; 
        else dummy = dummy->next;
        head = head->next;
    }
    return ans->next;
}

File: 3501-delete-nodes-from-linked-list-present-in-array/3501-delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        ListNode dummy = new ListNode();
        dummy.next = head;
        ListNode ans = dummy;
        boolean[] mp = new boolean[100001];
        for (int n: nums) mp[n] = true;
        while (head != null) {
            if (mp[head.val]) dummy.next = head.next;
            else dummy = dummy.next;
            head = head.next;
        }
        return ans.next;
    }
}